File: backend/src/main/java/entities/Project.java
package entities;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import io.quarkus.hibernate.orm.panache.PanacheEntity;

@Entity(name = "project")
@Table(name = "project")
public class Project extends PanacheEntity {

    public Project() {
    }

    public String number;
    public String name;
    public String description;

    @OneToMany(cascade = CascadeType.ALL, 
    mappedBy = "project", 
    fetch = FetchType.LAZY, 
    orphanRemoval = true)
    public List<Phase> data = new ArrayList<>();
}

File: src/app/shared/model/Project.ts
import { Phase, PhaseDto } from './Phase';
export class Project {
  constructor(
    public id: string,
    public name: string,
    public type: any,
    public description: string,
    public data: Phase[] = []
  ) {}
}

export interface ProjectDto {
  id: string;
  name: string;
  type: any;
  description: string;
  data: PhaseDto[];
}
